### DISCUSSION TITLE
unexpected error: supersetapierror: fatal error

### DISCUSSION THREAD

**Discussion by envicon** [apache/superset#23833] [Q&A / Help] [answered]
My Apache Superset installation has been running for at least 6 months without any changes.

Suddenly I get this error today, when I open a dashboard

"unexpected error: supersetapierror: fatal error"

I can login but can't view any dasboards or charts

When I open a chart I get this error:

{"errors": [{"message": "string argument without an encoding", "error_type": "GENERIC_BACKEND_ERROR", "level": "error", "extra": {"issue_codes": [{"code": 1011, "message": "Issue 1011 - Superset encountered an unexpected error."}]}}]}

I'm running Apache Superset 1.3.2 and Python 3.7.2 in an virtual environment

How to solve this issue?

**Comment by rikazfawaiz** [comment]
i have same issue, after login get this issue.

**Comment by JHNDaniel** [comment, marked as answer]
We solved the Encoding problem by changing sqlalchemy value for the class _Binary. 
- Apache Superset 1.3.2
- Python 3.7.2

[sqlalchemy/lib/sqlalchemy/sql/sqltypes.py](https://github.com/sqlalchemy/sqlalchemy/blob/5881fd274015af3de37f2ff0f91ff6a7c61c1540/lib/sqlalchemy/sql/sqltypes.py#L942:L948)
Line 942-948 function:
```
def result_processor(self, dialect, coltype):
       def process(value):
              if value is not None:
                      value = bytes(value)
              return value

       return process

```
The _Binary class changes are also used in _LargeBinary class, so with changing the value result_processor method under _Binary, you fix the encoding for both.
First I tried just changing the encoding to binary datatype in Line 945 (utf8, directly): 
```
value = bytes(value, 'utf8')
```
That did not work as intended and the problem kept appearing, so I fixed it by making some if statements instead, of the 945 single value line:
```
def result_processor(self, dialect, coltype):
       def process(value):
              if value is not None:
                      if isinstance(value, str):
                            value = bytes(value, 'utf8')
                      else:
                            value = bytes(value)
              return value

       return process

```
This solved our encoding problem, and superset is now working as intended again – have not appeared any bugs or other broken areas after the change to the sqlalchemy sqltypes.py file.

**Reply by felipe-alves-moraes** [reply]
Was this ever applied to a release in superset or sqlalchemy?
I'm facing the same issue with a pinot table that is saved as Byte.
